x/mongo/driver/connstring/connstring.go

```diff
@@ -37,7 +54,9 @@ type ConnString struct {
 	AppName                            string
 	AuthMechanism                      string
 	AuthMechanismProperties            map[string]string
+	AuthMechanismPropertiesSet         bool
```

Comment:
Are these changes needed? You can check if a map is set via `map != nil`.

Reply:
Yes because AuthMechanismProperties can be set to a default map in `parser.setDefaultAuthParams()`. I'm using `AuthMechanismPropertiesSet` to check if the `AuthMechanismProperties` had been set before we apply a default.

Reply:
Makes sense. Resolving.